parser-typechecker/src/Unison/Runtime/Builtin.hs

```diff
@@ -992,6 +992,19 @@ outIoFailNat stack1 stack2 stack3 fail nat result =
         $ TCon eitherReference 1 [nat])
   ]
 
+outIoFailInt :: forall v. Var v => v -> v -> v -> v -> v -> v -> ANormal v
+outIoFailInt stack1 stack2 stack3 fail int result =
+  TMatch result . MatchSum $ mapFromList
+  [ (0, ([BX, BX],)
+        . TAbss [stack1, stack2]
+        . TLetD fail BX (TCon Ty.failureRef 0 [stack1, stack2])
+        $ TCon eitherReference 0 [fail])
+  , (1, ([UN],)
+        . TAbs stack3
+        . TLetD int BX (TCon Ty.intRef 0 [stack3])
+        $ TCon eitherReference 1 [int])
+  ]
+
```

Comment:
Maybe rename this to `unitDirectInt` and simplify it.

@dolio do you know what this should be for a function `() -> IO Int`?
```

Reply:
I think in the naming scheme `Direct` means that it just returns a boxed value from the operation, but this is wrapping an unboxed int. Also this is only the part of the code that deals with the result, not the whole function.

I think this bit would just disappear, and the boxing would just be inlined into a renamed `unitToInt` `ForeignOp` or something.

Reply:
@dolio can you give `unitToInt :: ForeignOp` ?

Reply:
Instead of `outIoFailInt`, just use `TCon Ty.intRef 0 [result]`, I think.

Reply:
Done